app/models/stage.rb

```diff
@@ -177,6 +175,11 @@ def environments
     DeployGroup.enabled? ? deploy_groups.map(&:environment).uniq : []
   end
 
+  def command_ids=(value)
+    @command_ids_changed = (command_ids != value.map(&:to_i))
```

Comment:
if the ordering changed then the script will be modified, so we should record a new version